### ISSUE TITLE
Separate into its own module.

### ISSUE THREAD

**Issue by shahzeb1** [shahzeb1/soundcloud-player#3]
So right now [mySC.js](https://github.com/shahzeb1/soundcloud-player/blob/master/mySC.js) includes both the code for (a) interacting with SoundCloud (getting info / playing actual song) and (b) interacting with our UI (managing the play / pause button getting clicked.

(a) and (b) should be split up. (a) should be placed into its own `soundcloud-player.js` module. It should include the key functions which should allow for anyone to include the `<script>` and grab info / play music.

- [ ]  Separate `mySC.js` and `soundcloud-player.js` into two distinct files
- [ ]  Make it so that `soundcloud-player.js` includes all the soundcloud related functions
- [ ]  Write some unit tests for `soundcloud-player.js`
- [ ] Update the README.md with the appropriate changes as to how to use the API functions from the `soundcloud-player.js` module.

**Comment by R3l4x3** [comment]
Which testing suite do you want to use?

**Comment by shahzeb1** [comment]
That's a really good question. I don't have too much experience with any testing tool specifically, so I'm all ears. 👂 

Open to [any of these](https://stackshare.io/stackups/jasmine-vs-mocha-vs-qunit). I'm feeling QUnit because it handles our use-case pretty well, what about you?

**Comment by R3l4x3** [comment]
Okay, I'll take a look into this testing framework. 

At the moment I'm separating the player in its own file. And I have just following question: Should we remove jQuery completely?

Everything we do with jQuery is easy with VanillaJS. Or would you like to keep it?

**Comment by shahzeb1** [comment]
> Should we remove jQuery completely?

Yes that's actually a great idea. As far as I can tell you are correct. Everything will be possible via vanilla JS. 😄

**Comment by PaulTFreedman** [comment]
@shahzeb1 @R3l4x3 I'll probably start looking at this soon, let me know if I should hold off

**Comment by PaulTFreedman** [comment]
@shahzeb1 I just want to make sure I'm clear on the ultimate goal of this project - are you aiming to have a set of javascript functions that would allow someone to easily create their own SoundCloud players? I.e. is your UI here just to showcase an example of what your javascript functionality allows?

I'm wondering what assumptions could be made about how this would be used, for example will there have to be an iframe with id="so"?

**Comment by shahzeb1** [comment]
@PaulTFreedman yes so the goal is to separate all the Sound Cloud related logic into it's own library (file). This will make testing these specific functions easier. 

Anything related to the iFrame / Soundcloud should be placed in the new `soundcloud-player.js`

**Comment by R3l4x3** [comment]
@PaulTFreedman Feel free to looking at this 👍🏼

**Comment by PaulTFreedman** [comment]
I've made an attempt at the separation on my fork but haven't written any tests or updated the README yet. I'm not sure when I'll next have time to look at them so I'd be OK with someone taking over that part.

**Comment by shahzeb1** [comment]
@PaulTFreedman hey playa, feel free to make a fork request with the work you currently have. We'll forsure look it over and see what's going on / how we can help you with the tests! 🙌

**Comment by tehlordvortex** [comment]
I see @PaulTFreedman has separated the SoundCloud logic and UI logic. But this dumps everything in the global namespace. I would like to add some code that detects the current operating environment and exports the functions appropriately (Node.JS, RequireJS etc).

**Comment by PaulTFreedman** [comment]
@Vortex83 sounds good to me. I have started looking at writing unit tests using QUnit and Sinon but I can wait until you finish with that refactoring.
